CMakeLists.txt

```diff
@@ -44,7 +44,7 @@ if (EMSCRIPTEN)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -s TOTAL_STACK=5242880")
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -s TOTAL_STACK=5242880")
 
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-gpu-targets") # modified flag to avoid deprecated GPU targets warnings
```

Comment:
I think we should keep `-Wno-deprecated` as it is for general language features and not for CUDA specific warnings. 

And then we could add a check for `GGML_CUDA` and set the `-Wno-deprecated-gpu-targets` flag. The reason for this is that otherwise we will get warnings like the following on systems that don't support nvcc, for example on macos I get a number of warnings like the following:
```console
warning: unknown warning option '-Wno-deprecated-gpu-targets'; did you mean '-Wno-deprecated-pragma'? [-Wunknown-warning-option]
1 warning generated.
```